src/winetricks

```diff
@@ -6665,28 +6666,124 @@ load_dotnet46()
 
 #----------------------------------------------------------------
 
+w_metadata dotnet461 dlls \
+    title="MS .NET 4.6.1" \
+    publisher="Microsoft" \
+    year="2015" \
+    media="download" \
+    file1="NDP461-KB3102436-x86-x64-AllOS-ENU.exe" \
+    conflicts="dotnet20 dotnet20sdk dotnet20sp1 dotnet20sp2 dotnet35sp1 dotnet40 dotnet46 vjrun20" \
+    installed_file1="c:/windows/Microsoft.NET/Framework/v4.0.30319/SetupCache/v4.6.01055/1041/SetupResources.dll"
+
+load_dotnet461()
+{
+    if [ $W_ARCH = win64 ]
+    then
+        w_warn "This package may not work on a 64-bit installation"
+    fi
+
+    # https://www.microsoft.com/en-au/download/details.aspx?id=49982
+    w_download https://download.microsoft.com/download/E/4/1/E4173890-A24A-4936-9FC9-AF930FE3FA40/NDP461-KB3102436-x86-x64-AllOS-ENU.exe 83d048d171ff44a3cad9b422137656f585295866
+
+    w_call remove_mono
+
+    # Remove Mono registry entry:
+    "$WINE" reg delete "HKLM\Software\Microsoft\NET Framework Setup\NDP\v4" /f
+
+    w_try rm -f "$W_WINDIR_UNIX/system32/mscoree.dll"
+
+    w_call dotnet46
+    w_set_winver win7
+
+    w_try_cd "$W_CACHE/$W_PACKAGE"
+    
+    WINEDLLOVERRIDES=fusion=b "$WINE" "$file1" ${W_OPT_UNATTENDED:+/q /norestart}
+    status=$?
+    
+    echo "exit status: $status"
+
+    case $status in
+        0) ;;
+        105) echo "exit status $status - normal, user selected 'restart now'" ;;
+        194) echo "exit status $status - normal, user selected 'restart later'" ;;
+        *) w_die "exit status $status - $W_PACKAGE installation failed" ;;
+    esac
+
+    w_override_dlls native mscoree
+}
+
+#----------------------------------------------------------------
+
+w_metadata dotnet462 dlls \
+    title="MS .NET 4.6.2" \
+    publisher="Microsoft" \
+    year="2016" \
+    media="download" \
+    file1="dotNetFx462_Full_x86_x64_Slim.exe" \
+    conflicts="dotnet20 dotnet20sdk dotnet20sp1 dotnet20sp2 dotnet35sp1 dotnet40 dotnet46 dotnet461 vjrun20"
+
+load_dotnet462()
+{
+    if [ $W_ARCH = win64 ]
+    then
+        w_warn "This package may not work on a 64-bit installation"
+    fi
+
+    # Official .NET 4.6.2 offline installer fails. See https://bugs.winehq.org/show_bug.cgi?id=42170
+    # Un-official slim version does not fail. See http://repacks.net/forum/viewtopic.php?t=7
+    w_download http://repacks.net/uploads/dotNetFx462_Full_x86_x64_Slim.exe 1886c2b3b6eba37fcf79d59c9e5b100930150fda
```

Comment:
This needs to be behind a w_workaround_wine_bug check. It should then also support the official version, for once the bug is fixed.

I also suspect this problem may be staging specific..